cli/dcoscli/task/main.py

```diff
@@ -232,28 +230,27 @@ def _log(follow, completed, lines, task, file_):
                 raise DCOSException(msg)
         raise DCOSException('No matching tasks. Exiting.')
 
-    if file_ in ('stdout', 'stderr') and log.dcos_log_enabled():
-        try:
-            _dcos_log(follow, tasks, lines, file_, completed)
-            return 0
-        except (DCOSAuthenticationException,
-                DCOSAuthorizationException):
-            raise
-        except DCOSException as e:
-            emitter.publish(DefaultError(e))
-            emitter.publish(DefaultError('Falling back to files API...'))
-
-    mesos_files = _mesos_files(tasks, file_, client)
-    if not mesos_files:
-        if fltr is None:
-            msg = "No tasks found. Exiting."
-        else:
-            msg = "No matching tasks. Exiting."
-        raise DCOSException(msg)
-
-    log.log_files(mesos_files, follow, lines)
-
-    return 0
+    # if journald logging is disabled, read files API and exit.
+    if not log.dcos_log_enabled():
+        mesos_files = _mesos_files(tasks, file_, client)
+        if not mesos_files:
+            if fltr is None:
+                msg = "No tasks found. Exiting."
+            else:
+                msg = "No matching tasks. Exiting."
+            raise DCOSException(msg)
+
+        log.log_files(mesos_files, follow, lines)
+        return 0
+
+    # otherwise
+    if file_ in ('stdout', 'stderr'):
+        _dcos_log(follow, tasks, lines, file_, completed)
+        return 0
+
+    raise DCOSException('Only stdout/stderr files are supported by dcos-log '
```

Comment:
this will happen if dcos-log is enabled and user passes something like 
`dcos task log test.ae5b24da-f556-11e6-af80-0e47c9122ff4 myfile`